nemo/collections/nlp/data/question_answering_squad/qa_dataset.py

```diff
@@ -100,6 +109,11 @@ def __init__(
                 self.features = pickle.load(reader)
         else:
             logging.info(f"Preprocessing data.")
+            if num_samples == 0:
+                raise ValueError("num_samples has to be positive", num_samples)
```

Comment:
what about -1?

Reply:
num_samples = -1 is valid which takes the entire dataset, this is only to guard against user specifying num_samples=0 which is invalid

Reply:
maybe update the error message to say what -1 does?

Reply:
ok